packages/payments/src/sandbox-payments-repo.ts

```diff
@@ -0,0 +1,95 @@
+import type {PaymentsRepository, UnsubscribeCallback, UserSubscribedCallback} from "./payments-repo";
+import type { UserSubscription } from "./subscription";
+import {NoSubscriptionError} from "./errors";
+import {invariant, isProduction, warning} from "./utils";
+
+export type SandboxPaymentsOptions =
+  | boolean
+  | {
+  enabled?: boolean;
+  subscription: UserSubscription | null | (() => UserSubscription | null) | (() => Promise<UserSubscription | null>);
+};
+
+export class SandboxPaymentsRepo implements PaymentsRepository {
+  private readonly enabled: boolean;
+  private readonly _getSubscription: () => Promise<UserSubscription | null>;
+  private internalSubscription: UserSubscription | null = null;
+  private onUserSubscribedCallback: UserSubscribedCallback | null = null;
+
+  static isEnabled(options?: SandboxPaymentsOptions): boolean {
+    if (!options) {
+      return false;
+    }
+
+    const { enabled } = SandboxPaymentsRepo.parseOptions(options);
+    return enabled;
+  }
+
+  static parseOptions(options: SandboxPaymentsOptions): {
+    enabled: boolean;
+    getSubscription: () => Promise<UserSubscription | null>;
```

Comment:
```suggestion
    getSubscription: () => UserSubscription | null | Promise<UserSubscription | null>;
```

So that you can remove support of `subscription`

Reply:
Good idea, thanks!